core/src/types/voting_power.rs

```diff
@@ -169,6 +178,66 @@ impl BorshSchema for FractionalVotingPower {
     }
 }
 
+impl Serialize for FractionalVotingPower {
+    fn serialize<S>(
+        &self,
+        serializer: S,
+    ) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(&self.to_string())
+    }
+}
+
+struct VPVisitor;
+
+impl<'de> Visitor<'de> for VPVisitor {
+    type Value = FractionalVotingPower;
+
+    fn expecting(&self, formatter: &mut Formatter) -> std::fmt::Result {
+        formatter.write_str(
+            "A '/' separated pair of numbers, the second of which is non-zero.",
+        )
+    }
+
+    fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
+    where
+        E: de::Error,
+    {
+        let [numer, denom]: [&str; 2] =
+            value.split('/').collect::<Vec<&str>>().try_into().or(Err(
+                de::Error::custom("Expected a '/' separated pair of numbers"),
+            ))?;
+        let numer = numer
+            .trim()
+            .parse::<u64>()
+            .map_err(|e| de::Error::custom(e.to_string()))?;
+        let denom = denom
+            .trim()
+            .parse::<u64>()
+            .map_err(|e| de::Error::custom(e.to_string()))?;
```

Comment:
this technically allows you to parse `1/2` as well (without the whitespace). is that the intended behavior?

Reply:
actually reading through the code again, you're only making use of the serialization routine, right? deserialization doesn't seem to be used?

Reply:
That is intended behavior. And I added the deserialization because I figured I would use it in a test e2e test later.

Reply:
Although I could just add a `FromStr` impl later if I suppose.